homework/insensitive-palindrom/palindrome.cpp

```diff
@@ -0,0 +1,32 @@
+#include "palindrome.hpp"
+#include <algorithm>
+#include <cctype>
+#include <iostream>
+#include <iterator>
+
+std::string clean_string(const std::string& string) {
+    std::string cleaned;
+    std::copy_if(begin(string),
+                 end(string),
+                 std::back_inserter(cleaned),
+                 [](auto sign) {
+                     return ('A' <= sign && sign <= 'Z') || ('a' <= sign && sign <= 'z');
+                 });
+    std::transform(begin(cleaned),
+                   end(cleaned),
+                   begin(cleaned),
+                   [](auto sign) {
+                       return std::tolower(sign);
+                   });
+    return cleaned;
+}
+
+bool is_palindrome(const std::string& string) {
+    std::string cleaned = clean_string(string);
+    int length = cleaned.size();
+    int middle = length / 2;
+    std::string first = cleaned.substr(0, middle);
+    std::string second = cleaned.substr(length - middle, length);
```

Comment:
This work, but also is not efficient. Better option is to compare stirng from two edge. Use for isntanc eequal and pass iterator to begin and reversed iterator. Also you need to comapre only half string :)